From d3e2854d7f79d5b7ddb01b9ff940fa9729b8b7bd Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Fri, 4 Apr 2014 10:40:47 -0400
Subject: [PATCH] Include multiboot module parameters in --info (#997934)

Resolves: rhbz#997934

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grubby.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/grubby.c b/grubby.c
index a5637d3..065448c 100644
--- a/grubby.c
+++ b/grubby.c
@@ -1584,6 +1584,7 @@ void displayEntry(struct singleEntry * entry, const char * prefix, int index) {
     struct singleLine * line;
     char * root = NULL;
     int i;
+    int j;
 
     printf("index=%d\n", index);
 
@@ -1657,6 +1658,20 @@ void displayEntry(struct singleEntry * entry, const char * prefix, int index) {
 	    printf("%s%s", line->elements[i].item, line->elements[i].indent);
 	printf("\n");
     }
+
+    for (j = 0, line = entry->lines; line; line = line->next) {
+	if ((line->type & LT_MBMODULE) && line->numElements >= 2) {
+	    if (!strncmp(prefix, line->elements[1].item, strlen(prefix)))
+		printf("mbmodule%d=", j);
+	    else
+		printf("mbmodule%d=%s", j, prefix);
+
+	    for (i = 1; i < line->numElements; i++)
+		printf("%s%s", line->elements[i].item, line->elements[i].indent);
+	    printf("\n");
+	    j++;
+	}
+    }
 }
 
 int parseSysconfigGrub(int * lbaPtr, char ** bootPtr) {
-- 
1.9.0

